x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/policy_json_flyout.tsx

```diff
@@ -45,26 +46,28 @@ export const PolicyJsonFlyout: React.FunctionComponent<Props> = ({
   const [policy, setPolicy] = useState<undefined | null | SerializedPolicy>(undefined);
 
   const form = useFormContext();
-  const [formData, getFormData] = useFormData();
+  const [, getFormData] = useFormData<FormInternal>();
+
+  const getPolicy = useCallback(async () => {
+    setPolicy(undefined);
+    if (await form.validate()) {
+      const p = getFormData() as SerializedPolicy;
+      setPolicy({
+        ...legacyPolicy,
+        phases: {
+          ...legacyPolicy.phases,
+          hot: p.phases.hot,
+          warm: p.phases.warm,
+        },
+      });
+    } else {
+      setPolicy(null);
+    }
+  }, [setPolicy, getFormData, legacyPolicy, form]);
 
   useEffect(() => {
-    (async function checkPolicy() {
-      setPolicy(undefined);
-      if (await form.validate()) {
-        const p = getFormData() as SerializedPolicy;
-        setPolicy({
-          ...legacyPolicy,
-          phases: {
-            ...legacyPolicy.phases,
-            hot: p.phases.hot,
-          },
-        });
-      } else {
-        setPolicy(null);
-      }
-    })();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [form, legacyPolicy, formData]);
+    getPolicy();
```

Comment:
`getPolicy` or `updatePolicy`? 😊 


Also it would be better to deconstruct the `validate` handler from `form` to not pass the whole object.